packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -989,15 +989,7 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 
 	function updateNodeAtIndex(nodeIndex: number, nodeData: Partial<INodeUi>): void {
 		if (nodeIndex !== -1) {
-			const node = workflow.value.nodes[nodeIndex];
-			workflow.value = {
-				...workflow.value,
-				nodes: [
-					...workflow.value.nodes.slice(0, nodeIndex),
-					{ ...node, ...nodeData },
-					...workflow.value.nodes.slice(nodeIndex + 1),
-				],
-			};
+			workflow.value.nodes[nodeIndex] = { ...workflow.value.nodes[nodeIndex], ...nodeData };
```

Comment:
This doesn't update the node it replaces it, could please try the aproach below?
```suggestion
			Object.assign(workflow.value.nodes[nodeIndex], nodeData)
```

Reply:
Great suggestion! Done.